### ISSUE TITLE
RenameKeywords formatting rule should not rename embedded variable values to Title Case

### ISSUE THREAD

**Issue by apallier** [MarketSquare/robotframework-robocop#1627]
## Environment:

Robocop version: 7.2.0
OS: Ubuntu 24.04.1
Python version: 3.12.3

## Steps to reproduce:

1. Create a Robot Framework file (`my_file.robot`) with a keyword containing an embedded variable with spaces in its value, for example:

```robotframework
*** Settings ***
Library     String

*** Test Cases ***
Test
    Keyword With "my variable string with spaces"

*** Keywords ***
Keyword With "${embedded_variable}"
    Should Be Lower Case    ${embedded_variable}
```

2. Run Robotidy with the RenameKeywords transformer enabled:

```shell
robocop format --select RenameKeywords --configure RenameKeywords.enabled=True my_file.robot
```

## Actual Behavior:

The test doesn't pass anymore:

```shell
$ robot my_file.robot
==============================================================================
My File                                         
==============================================================================
Test                                                                  | FAIL |
'my Variable String With Spaces' is not lower case.
------------------------------------------------------------------------------
My File                                  | FAIL |
1 test, 0 passed, 1 failed
==============================================================================
```

The new file content is:

```robotframework
*** Settings ***
Library     String

*** Test Cases ***
Test
    Keyword With "my Variable String With Spaces"

*** Keywords ***
Keyword With "${embedded_variable}"
    Should Be Lower Case    ${embedded_variable}
```

## Expected Behavior:

Embedded variable values should never be modified by RenameKeywords because it could change the code behavior.

Only keyword name parts should be transformed to Title Case.

## Remarks

- The [doc includes a "warning"](https://robocop.dev/stable/formatter/formatters/RenameKeywords/#renamekeywords) on the "RenameKeywords" usage with embedded variables:

  > It has only basic support for keywords with embedded variables - use it at your own risk.

  But this behavior seems to be a "regression" because ever discussed in the original tool Robotidy here: [Do not rename embedded variables in RenameKeywords #428](https://github.com/MarketSquare/robotframework-tidy/pull/428)

- May be related issues:

  - https://github.com/MarketSquare/robotframework-robocop/issues/1555
  - https://github.com/MarketSquare/robotframework-robocop/issues/1169

**Comment by bhirsz** [comment]
The linked issue either solved different issue or there was no fix. As for Robotidy story, it focused on recognizing the variables ("My ${variable_inside}"). There is some similarity because becasue we split on ", incorrect part of the keyword was recognized as keyword name.

As you may know, Robocop does not know if keyword is embedded keyword or not. But since you use popular convenienc - quoting the embedded values - we can use it as basis for recognition. So we could ignore values inside quotes ("") and pass it as is. What do you think?
